packages/core/src/NavigationFocusedContext.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { NavigationHelpers, ParamListBase } from './types';
+
+export type FocusedNavigationCallback = (
+  navigation: NavigationHelpers<ParamListBase>
+) => any;
+
+export type FocusedNavigationListener = (
+  callback: FocusedNavigationCallback
+) => { handled: boolean; result: any };
+
+const MISSING_CONTEXT_ERROR =
+  "We couldn't find a navigation focused context. Have you wrapped your app with 'NavigationContainer'?";
```

Comment:
Maybe we should't throw this error? A "navigation focused context" is an internal thing, so the error message could be confusing for the user

Reply:
I think that it's not important. I just didn't want to make keys optional.

Reply:
maybe I can change default to: `() => {}`

Reply:
Or whatever, it can be optional. Fixed!